apps/sms/test/unit/thread_ui_test.js

```diff
@@ -3586,48 +3683,52 @@ suite('thread_ui.js >', function() {
           subject:'',
           attachments: null,
           timestamp: Date.now()
-        });
-
-        // Retrieve the message node
-        var messageNode = document.querySelector('#message-12 section');
-
-        // Dispatch custom event for testing long press
-        messageNode.dispatchEvent(contextMenuEvent);
-        assert.ok(MockOptionMenu.calls.length, 1);
-        assert.ok(MockOptionMenu.calls[0].items.length, 4);
-
-        // Confirm that the menu contained a "resend-message" option
-        for (var i = MockOptionMenu.calls[0].length - 1; i >= 0; i--) {
-          assert.notEqual(
-            MockOptionMenu.calls[0].items[i].l10nId,
-            'forward'
-          );
-        }
+        }).then(() => {
+          // Retrieve the message node
+          var messageNode = document.querySelector('#message-12 section');
+
+          // Dispatch custom event for testing long press
+          messageNode.dispatchEvent(contextMenuEvent);
+          assert.ok(MockOptionMenu.calls.length, 1);
+          assert.ok(MockOptionMenu.calls[0].items.length, 4);
+
+          // Confirm that the menu contained a "resend-message" option
+          for (var i = MockOptionMenu.calls[0].length - 1; i >= 0; i--) {
+            assert.notEqual(
+              MockOptionMenu.calls[0].items[i].l10nId,
+              'forward'
+            );
+          }
+        }).then(done, done);
     });
   });
 
   suite('Message resending UI', function() {
-    setup(function() {
-      ThreadUI.appendMessage({
+    setup(function(done) {
+      var promises = [];
+
+      promises.push(ThreadUI.appendMessage({
         id: 23,
         type: 'sms',
         body: 'This is a test',
         delivery: 'error',
         timestamp: Date.now()
-      });
-      ThreadUI.appendMessage({
+      }), ThreadUI.appendMessage({
         id: 45,
         type: 'sms',
         body: 'This is another test',
         delivery: 'sent',
         timestamp: Date.now()
-      });
-      this.sinon.stub(Utils, 'confirm').returns(Promise.resolve());
-      this.sinon.stub(ThreadUI, 'resendMessage');
-      this.elems = {
-        errorMsg: ThreadUI.container.querySelector('.error'),
-        sentMsg: ThreadUI.container.querySelector('.sent')
-      };
+      }));
+
+      Promise.all(promises).then(() => {
+        this.sinon.stub(Utils, 'confirm').returns(Promise.resolve());
```

Comment:
nit: I guess no need in defining stubs inside ```then``` callback, but I'm fine with it :)